File: internal/output/style.go
package output

import (
	"fmt"
	"os"
	"strings"
	"sync/atomic"
)

var colorsEnabled atomic.Bool

func init() {
	Configure(false)
}

// Configure updates terminal styling behavior from flags and environment.
func Configure(noColor bool) {
	enabled := !noColor

	if os.Getenv("NO_COLOR") != "" {
		enabled = false
	}

	if strings.EqualFold(os.Getenv("TERM"), "dumb") {
		enabled = false
	}

	if os.Getenv("CLICOLOR_FORCE") == "1" {
		enabled = true
	}

	colorsEnabled.Store(enabled)
}

// SetColorEnabled force-enables or disables ANSI styling.
func SetColorEnabled(enabled bool) {
	colorsEnabled.Store(enabled)
}

func style(code, text string) string {
	if text == "" || !colorsEnabled.Load() {
		return text
	}

	return fmt.Sprintf("\x1b[%sm%s\x1b[0m", code, text)
}

// Bold makes text bold.
func Bold(text string) string {
	return style("1", text)
}

// Muted dims text.
func Muted(text string) string {
	return style("2", text)
}

// Accent highlights text with cyan.
func Accent(text string) string {
	return style("36", text)
}

// Success highlights successful text.
func Success(text string) string {
	return style("32;1", text)
}

// Failure highlights failed text.
func Failure(text string) string {
	return style("31;1", text)
}

// Warning highlights warning text.
func Warning(text string) string {
	return style("33;1", text)
}

// Info highlights informational text.
func Info(text string) string {
	return style("34;1", text)
}

File: src/output/style.rs
use std::sync::atomic::{AtomicBool, Ordering};

static COLORS_ENABLED: AtomicBool = AtomicBool::new(true);

pub fn configure(no_color: bool) {
    let mut enabled = !no_color;

    if std::env::var_os("NO_COLOR").is_some() {
        enabled = false;
    }

    if let Ok(term) = std::env::var("TERM")
        && term.eq_ignore_ascii_case("dumb")
    {
        enabled = false;
    }

    if std::env::var("CLICOLOR_FORCE").ok().as_deref() == Some("1") {
        enabled = true;
    }

    COLORS_ENABLED.store(enabled, Ordering::Relaxed);
}

fn style(code: &str, text: &str) -> String {
    if text.is_empty() || !COLORS_ENABLED.load(Ordering::Relaxed) {
        return text.to_string();
    }

    format!("\x1b[{code}m{text}\x1b[0m")
}

pub fn bold(text: &str) -> String {
    style("1", text)
}

pub fn muted(text: &str) -> String {
    style("2", text)
}

pub fn accent(text: &str) -> String {
    style("36", text)
}

pub fn success(text: &str) -> String {
    style("32", text)
}

pub fn failure(text: &str) -> String {
    style("31", text)
}

pub fn warning(text: &str) -> String {
    style("33", text)
}

pub fn info(text: &str) -> String {
    style("96", text)
}

pub fn command(text: &str) -> String {
    style("96", text)
}

pub fn number(text: &str) -> String {
    style("96", text)
}

pub fn bullet(text: &str) -> String {
    style("94", text)
}
